src/contracts/conquest/ConquestV2.sol

```diff
@@ -0,0 +1,272 @@
+pragma solidity 0.7.4;
+
+import "../utils/TieredOwnable.sol";
+import "../interfaces/ISkyweaverAssets.sol";
+import "../interfaces/IRewardFactory.sol";
+import "../interfaces/IConquest.sol";
+import "@0xsequence/erc-1155/contracts/utils/SafeMath.sol";
+import "@0xsequence/erc-1155/contracts/interfaces/IERC165.sol";
+import "@0xsequence/erc-1155/contracts/interfaces/IERC1155.sol";
+import "@0xsequence/erc-1155/contracts/interfaces/IERC1155TokenReceiver.sol";
+
+/**
+ * @notice Keep track of players participation in Conquest and used to issue rewards.
+ * @dev This contract must be at least a TIER 1 owner of the silverCardFactory and
+ *      goldCardFactory.
+ *
+ *      Differences with V1: 
+ *        - No timestamp check (was unused)
+ *        - Exiting conquest validates the conquest nonce (to prevent race conditions)
+ *        - Checks V1 state for brand new conquest users, to ensure continuity
+ */
+contract ConquestV2 is IERC1155TokenReceiver, TieredOwnable {
+  using SafeMath for uint256;
+
+  /***********************************|
+  |             Variables             |
+  |__________________________________*/
+
+  // Parameters
+  uint256 constant internal ENTRIES_DECIMALS = 2; // Amount of decimals conquest entries have
+  uint256 constant internal CARDS_DECIMALS = 2;   // Number of decimals cards have
+
+  // Contracts
+  IConquest immutable public oldConquest;            // Old conquest contract needed for existing states
+  IRewardFactory immutable public silverCardFactory; // Factory that mints Silver cards
+  IRewardFactory immutable public goldCardFactory;   // Factory that mints Gold cards
+  ISkyweaverAssets immutable public skyweaverAssets; // ERC-1155 Skyweaver assets contract
+  uint256 immutable public conquestEntryID;          // Conquest entry token id
+
+  // Mappings
+  mapping(address => bool) internal _isActiveConquest;    // Whether a given player is currently in a conquest
+  mapping(address => uint256) internal _conquestsEntered; // Number of conquest a given player has entered so far
```

Comment:
Does making these `internal` mean that the `syncState` trick won't work if we have to do a `Conquest V3` in the future?

Reply:
Never mind, I didn't finish reading the PR :P